app/products/agents/store/streaming_store.ts

```diff
@@ -0,0 +1,265 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {CONTROL_SIGNALS} from '@agents/constants';
+import {StreamingEvents, type StreamingState, type PostUpdateWebsocketMessage} from '@agents/types';
+import {DeviceEventEmitter} from 'react-native';
```

Comment:
Wondering if we could have this store use observables instead of emitting events? Did you look into that?

Reply:
ok not a blocker.. but it would be nice to explore the use of observables instead of emitting events.

My reason for this is that observables are better for state management while `DeviceEventEmitter` is mostly to communicate between native and JS event bridges, sure it works here, but I think its worth exploring. You can check how it was done with `@products/calls/state/*`